medicines/doc-index-updater/src/auth_manager/mod.rs

```diff
@@ -0,0 +1,136 @@
+extern crate base64;
+
+use futures::future;
+use regex::Regex;
+use warp::{Filter, Rejection};
+
+#[derive(Clone, Debug)]
+pub struct AuthenticationFailed;
+impl warp::reject::Reject for AuthenticationFailed {}
+
+fn get_basic_username() -> String {
+    std::env::var("BASIC_AUTH_USERNAME").expect("Set env variable BASIC_AUTH_USERNAME first!")
```

Comment:
doesn't need to be done this PR, but I think we should be running these env var checks at main.rs unless we expect them to change mid-run. I wanna know they're missing as soon as possible

Reply:
Agreed, we should capture this in a ticket because a lot of our env isn't being checked at runtime.

Reply:
https://github.com/MHRA/products/issues/535